python/docs/Makefile

```diff
@@ -1,19 +1,35 @@
 # Makefile for Sphinx documentation
 #
 
+ifndef SPHINXPYTHON
+SPHINXBUILD ?= sphinx-build
+# User-friendly check for sphinx-build if explicitly specified.
+ifeq ($(shell which $(SPHINXBUILD) >/dev/null 2>&1; echo $$?), 1)
+$(error The '$(SPHINXBUILD)' command was not found. Make sure you have Sphinx installed, then set the SPHINXBUILD environment variable to point to the full path of the '$(SPHINXBUILD)' executable. Alternatively you can add the directory with the executable to your PATH. If you don't have Sphinx installed, grab it from http://sphinx-doc.org/)
+endif
```

Comment:
If `SPHINXPYTHON -msphinx' is preferred instead of 'sphinx-build'` by default, should we go for a default `SPHINXPYTHON` + `-msphinx` instead of a default `SPHINXBUILD` if both are not set?

Reply:
Yea... that was the problem I faced.. Probably we should change but seems it needs some changes in Jenkins environment .. FYI there are some discussion about the potential effect by setting that as a default in the PR link of the comments.

I was thinking we set `SPHINXPYTHON` manually when we make a release for now.

Reply:
Maybe we should let `SPHINXBUILD` have higher priority for now. Let me update later.

Reply:
That seems error prone - why not pick a good default of python3 and allow overriding?